### PR TITLE
refactor(payment_methods): Fix db insert for payment method create

### PR DESCRIPTION
**Pull request by Sarthak1799**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Currently, payment method create API does not insert payment method record in the Db. 
have added the db call to insert the record if not already present.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

image

image

image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/payment_methods/cards.rs [hunk 1]
```diff
@@ -108,7 +114,24 @@ pub async fn add_payment_method(
...
     };
-    Ok(response?.0).map(services::ApplicationResponse::Json)
+
+    let (resp, is_duplicate) = response?;
+    if !is_duplicate {
+        let pm_metadata = resp.metadata.as_ref().map(|data| data.clone().expose());
+        create_payment_method(
+            &*state.store,
+            &req,
+            &customer_id.clone(),
```
Comment (pixincreate) [inline crates/router/src/core/payment_methods/cards.rs:124]:
```suggestion
            &customer_id,
```

**Review by pixincreate — COMMENTED** [review]
Apart from that, looks good to me.

**Review by pixincreate — DISMISSED** [review]

File: crates/router/src/core/payment_methods/cards.rs [hunk 2]
```diff
@@ -108,7 +114,24 @@ pub async fn add_payment_method(
             Ok((payment_method_response, false))
         }
     };
-    Ok(response?.0).map(services::ApplicationResponse::Json)
+
+    let (resp, is_duplicate) = response?;
+    if !is_duplicate {
+        let pm_metadata = resp.metadata.as_ref().map(|data| data.clone().expose());
```
Comment (pixincreate) [inline crates/router/src/core/payment_methods/cards.rs:120]:
```suggestion
        let pm_metadata = resp.metadata.as_ref().map(|data| data.peek());
```

and use 
`pm_metadata.cloned()` in line 127

**Review by Narayanbhat166 — APPROVED** [review]

**Review by pixincreate — DISMISSED** [review]

File: CHANGELOG.md
Comment (SanchithHegde) [inline CHANGELOG.md:1]: Please revert changes to this file.

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Review by SanchithHegde — APPROVED** [review]

**Review by pixincreate — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**